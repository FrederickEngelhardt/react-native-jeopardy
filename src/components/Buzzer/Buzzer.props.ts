const buzzerProps = {
    title: "Buzzer"
};

export default buzzerProps;
