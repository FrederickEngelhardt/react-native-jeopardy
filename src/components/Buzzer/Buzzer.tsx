import React from "react";
import styled from "styled-components/native";

import { black, grey, white, yellow } from "../../constants/theming";
import { TitleText } from "../Card/Card";
import { Theme } from "../../containers/RootThemeProvider";

const BuzzerBox = styled.View`
  background-color: ${({ theme }) => (theme.darkMode ? yellow : white)};
  width: 85;
  height: 85;
  align-items: center;
  justify-content: center;
  border-radius: 25px;
`;
const BuzzerButton = styled.TouchableHighlight.attrs(props => ({
  underlayColor: black
}))`
  background-color: ${({ theme }) => (theme.darkMode ? white : grey)};
  border-color: black;
  border-width: 3px;
  border-radius: 50px;
  height: 75;
  width: 75;
  justify-content: center;
`;

const BuzzerText = styled(TitleText)`
  font-size: 18px;
  color: ${props => (props.theme.darkMode ? black : black)};
`;

interface Props {
  handleClick?: () => void;
  theme?: Theme;
  title?: string;
}

const Buzzer = ({ title, handleClick }: Props) => {
  return (
    <BuzzerBox>
      <BuzzerButton
        accessible
        acessibilityLabel="Jeopardy Buzzer"
        onPress={handleClick}
      >
        <BuzzerText>{title}</BuzzerText>
      </BuzzerButton>
    </BuzzerBox>
  );
};

Buzzer.defaultProps = {
  title: "Buzzer"
};

export default Buzzer;
