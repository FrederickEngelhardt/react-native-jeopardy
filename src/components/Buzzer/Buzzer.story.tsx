import { storiesOf } from "@storybook/react-native";
import * as React from "react";
import { withKnobs } from "@storybook/addon-knobs";
import {action} from "@storybook/addon-actions";

import Buzzer from "./Buzzer";
import buzzerProps from "./Buzzer.props";

storiesOf("<Buzzer />", module)
  .addDecorator(withKnobs)
  .add("default", () => <Buzzer {...buzzerProps} handleClick={action('clicked buzzer')} />);
