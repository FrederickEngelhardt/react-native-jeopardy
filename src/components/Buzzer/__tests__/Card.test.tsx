import React from "react";
import { render } from "@testing-library/react-native";

import Buzzer from "../Buzzer";
import buzzerProps from "../Buzzer.props";

const theme = {
  darkMode: true,
  deviceHeight: 100,
  deviceWidth: 100
};

const handleClick = jest.fn();

describe("<Buzzer />", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Buzzer {...buzzerProps} theme={theme} handleClick={() => handleClick} />
    );
    expect(container).toMatchSnapshot();
  });
});
