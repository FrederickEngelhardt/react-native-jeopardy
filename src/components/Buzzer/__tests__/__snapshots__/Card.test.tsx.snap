// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Buzzer /> renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 25,
          "height": 85,
          "justifyContent": "center",
          "width": 85,
        },
      ]
    }
  >
    <TouchableHighlight
      accessible={true}
      acessibilityLabel="Jeopardy Buzzer"
      activeOpacity={0.85}
      delayPressOut={100}
      style={
        Array [
          Object {
            "backgroundColor": "#d1d1d1",
            "borderColor": "black",
            "borderRadius": 50,
            "borderWidth": 3,
            "height": 75,
            "justifyContent": "center",
            "width": 75,
          },
        ]
      }
      underlayColor="#000"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 18,
              "fontWeight": "600",
              "textAlign": "center",
            },
          ]
        }
      >
        Buzzer
      </Text>
    </TouchableHighlight>
  </View>
</View>
`;
